Smooth scrolling animation and configuration

Wdestroier opened the pull request:
Smooth scrolling is a nice to have feature. It can enhance the experience for some users and contribute to a more polished and professional feel.

I checked Flutter's `DataTable` widget, the `two_dimensional_scrollables`, `data_table_2`, `material_table_view`, `trina_grid`, `flutter_web_data_table` and `syncfusion_flutter_datagrid` packages, but none of them have smooth scrolling.

I implemented smooth scrolling for this package, because it's the most complete and free (unlike syncfusion) grid/table widget I could find.

I kept it disabled by default, but it can be enabled in the configuration:
```dart
child: TrinaGrid(
  configuration: const TrinaGridConfiguration(
    scrollbar: TrinaGridScrollbarConfig(
      smoothScrolling: true,
    ),
  ),
  ...
),
```

Recording showing the feature disabled and enabled:


https://github.com/user-attachments/assets/b08606c1-b14c-45a8-8828-967a99128b10

mateusfccp commented on lib/src/ui/trina_body_rows.dart:
```diff
@@ -224,7 +229,14 @@ class TrinaBodyRowsState extends TrinaStateWithChange<TrinaBodyRows> {
                             ),
                           // Scrollable rows
                           Expanded(
+<<<<<<< Updated upstream
                             child: ListView.builder(
+=======
+                            child: (scrollConfig.smoothScrolling
+                                ? TrinaSmoothListView.builder
+                                : ListView.builder)(
+                              cacheExtent: stateManager.rowsCacheExtent,
+>>>>>>> Stashed changes
```
It seems this is an unresolved conflict.

Wdestroier replied:
Thank you. I merged the changes, hopefully PR #76 is accepted too.

doonfrs commented:
This is great !
I will merge it and check if we can replace it with the current scroll, if all the features are implemented.

doonfrs commented:
Hi @Wdestroier 
I am sorry for reversing your commit, it is awesome actually, I planned to use it as default.
But actually, I used the latest table, and still can not build, it only worked on the master version and it is not good to use the master as it is not the standard version, at least for a package.
I don't know if I was wrong, maybe I did not use the latest stable version?

Wdestroier commented:
Was the project not compiling? I merged both changes in a single branch and I'm currently importing it in another project with:
```
dependencies:s
  trina_grid:
    git:
      url: https://github.com/wdestroier/trina_grid.git
      ref: cache-and-scroll
```

doonfrs commented:
git checkout 6c2b44ac85a9eb747def741f460683d77e8e5847
flutter --version
Flutter 3.29.2 • channel stable • https://github.com/flutter/flutter.git
Framework • revision c236373904 (3 months ago) • 2025-03-13 16:17:06 -0400
Engine • revision 18b71d647a
Tools • Dart 3.7.2 • DevTools 2.42.3

![image](https://github.com/user-attachments/assets/be3c1a8a-ca2f-4715-a95f-6dbe7e204799)

doonfrs commented:
Now it worked after flutter upgrade to the latest veresion, but I think this feature is prettry new ( < 2 months ) , I am afraid that enforcing to use the latest flutter from the package will not be a good idea.

let's wait 3 months at least then we merge it ... what do you think?

doonfrs commented:
I tried this actually; unfortunately, pub will use the current installed version, and the build will fail, so it's better to wait, and then we will use the smooth scroll instead of the current scroll version—it is way better.

Wdestroier commented:
After these months I found and fixed bug that the Timer could finish after the widget is disposed. An error message was rarely printed in the console. I added a check to fix the problem in the https://github.com/Wdestroier/trina_grid/commit/c7751c73e7fc75b735caaf03e26a86a7a92a43be commit. I didn't find the original Timer code after the refactoring. @doonfrs, do you know if the problem might still happen after the enhancements?
